context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { useEventCallback } from '@mui/material/utils';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import AddRounded from '@mui/icons-material/AddRounded';
+
+import SelectableTableProvider from 'js/shared-styles/tables/SelectableTableProvider';
+import { WorkspacesIcon } from 'js/shared-styles/icons';
+
+import { useOpenDialog } from '../../workspaces/WorkspacesDropdownMenu/WorkspacesDropdownMenu';
+import { useCreateWorkspaceForm } from '../../workspaces/NewWorkspaceDialog/useCreateWorkspaceForm';
+import NewWorkspaceDialog from '../../workspaces/NewWorkspaceDialog/NewWorkspaceDialog';
+import AddDatasetsFromSearchDialog from '../../workspaces/AddDatasetsFromSearchDialog/AddDatasetsFromSearchDialog';
+import { useAppContext, useFlaskDataContext } from '../../Contexts';
+import { useTrackEntityPageEvent } from '../useTrackEntityPageEvent';
+
+interface ProcessedDataWorkspaceMenuProps {
+  button: React.ReactNode;
+}
+
+function ProcessedDataWorkspaceMenu({ button }: ProcessedDataWorkspaceMenuProps) {
+  const {
+    entity: { mapped_data_access_level, hubmap_id, uuid, entity_type, status },
+  } = useFlaskDataContext();
+  const isDataset = entity_type === 'Dataset';
+  const { isWorkspacesUser } = useAppContext();
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const track = useTrackEntityPageEvent();
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    track({
+      action: 'Open Workspace Menu',
+      label: hubmap_id,
+    });
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const {
+    control,
+    errors,
+    removeDatasets,
+    setDialogIsOpen: setOpenCreateWorkspace,
+    dialogIsOpen: createWorkspaceIsOpen,
+    ...rest
+  } = useCreateWorkspaceForm({
+    defaultName: hubmap_id,
+    initialSelectedDatasets: [uuid] ?? [],
```

Comment:
The `uuid` and `hubmap_id` will always be the primary dataset since it comes from the flask data context. Doesn't the helper panel only appear for the current processed dataset in view?

Reply:
Yes - there is a zustand store that provides the current visible helper panel content.

Also, [uuid] is never nullish due to list wrapper; if we want to fall back to an empty array in this case, [uuid].filter(Boolean) should work.

Reply:
Got it- updated this to reflect the current processed dataset.